modin/pandas/series.py

```diff
@@ -718,9 +718,9 @@ def between(self, left, right, inclusive: str = "both"):  # noqa: PR01, RT01, D2
         """
         Return boolean Series equivalent to left <= series <= right.
         """
-        return self.__constructor__(
-            query_compiler=self._query_compiler.between(left, right, inclusive)
-        )
+        # 'pandas.Series.between()' only uses public Series' API,
+        # so passing a Modin Series there is safe
+        return pandas.Series.between(self, left, right, inclusive)
```

Comment:
I wouldn't rely on pandas to handle Modin objects properly. We should probably pass a pandas Series or a numpy array in to the method.